pkg/component/shoot/system/system.go

```diff
@@ -140,7 +140,35 @@ func (s *shootSystem) WaitCleanup(ctx context.Context) error {
 }
 
 func (s *shootSystem) computeResourcesData() (map[string][]byte, error) {
-	registry := managedresources.NewRegistry(kubernetes.ShootScheme, kubernetes.ShootCodec, kubernetes.ShootSerializer)
+	var (
+		registry              = managedresources.NewRegistry(kubernetes.ShootScheme, kubernetes.ShootCodec, kubernetes.ShootSerializer)
+		k8sGreaterEqual133, _ = versionutils.CheckVersionMeetsConstraint(s.values.KubernetesVersion.String(), ">= 1.33")
```

Comment:
Look at https://github.com/gardener/gardener/blob/dc79b4fba2328fa7c590c2c126a18083e6548d69/pkg/component/shoot/system/system.go#L378 - this is how such checks are made